tgui/packages/tgui/interfaces/NtosMessenger/index.tsx

```diff
@@ -0,0 +1,336 @@
+import { Box, Button, Icon, Section, Stack, Input, TextArea, Dimmer, Divider } from '../../components';
+import { useBackend, useLocalState } from '../../backend';
+import { createSearch } from 'common/string';
+import { BooleanLike } from 'common/react';
+import { NtosWindow } from '../../layouts';
+
+import { NtChat, NtMessenger, NtPicture } from './types';
+import { ChatScreen } from './ChatScreen';
+import { sortBy } from 'common/collections';
+
+type NtosMessengerData = {
+  can_spam: BooleanLike;
+  is_silicon: BooleanLike;
+  owner?: NtMessenger;
+  saved_chats: Record<string, NtChat>;
+  messengers: Record<string, NtMessenger>;
+  sort_by_job: BooleanLike;
+  alert_silenced: BooleanLike;
+  alert_able: BooleanLike;
+  sending_and_receiving: BooleanLike;
+  open_chat: string;
+  stored_photos?: NtPicture[];
+  selected_photo_path?: string;
+  on_spam_cooldown: BooleanLike;
+  virus_attach: BooleanLike;
+  sending_virus: BooleanLike;
+};
+
+export const NtosMessenger = (_: any, context: any) => {
```

Comment:
implied any's, no need to type this
also I got fire before for underscoring to deal with the unused variable. it's more important to leave in so that others understood what went there. 
`(props, context)` please, and same for the others